frontend/public/components/masthead.tsx

```diff
@@ -9,12 +10,96 @@ import * as dedicatedLogoImg from '../imgs/openshift-dedicated-logo.svg';
 import { FLAGS, connectToFlags, flagPending } from '../features';
 import { authSvc } from '../module/auth';
 import { Dropdown, ActionsMenu } from './utils';
+import { SoftwareDetails } from './software-details';
 
 import { coFetchJSON } from '../co-fetch';
 import { SafetyFirst } from './safety-first';
 
 const developerConsoleURL = (window as any).SERVER_FLAGS.developerConsoleURL;
 
+export class HelpMenu extends React.Component<HelpMenuProps, HelpMenuState> {
+  constructor (props) {
+    super(props);
+    this.state = { showModal: false };
+    this.close = this.close.bind(this);
+    this.open = this.open.bind(this);
+    this.redirect = this.redirect.bind(this);
+  }
+
+  close() {
+    this.setState({ showModal: false });
+  }
+
+  open() {
+    this.setState({ showModal: true });
+  }
+
+  redirect() {
+    window.open('https://docs.openshift.com/online/welcome/index.html', '_blank');
+  }
+
+  render() {
+    const {username} = this.props;
+    return <React.Fragment>
+      <ActionsMenu
+        actions={[
+          {label: 'Documentation', callback: this.redirect},
+          {label: 'About', callback: this.open}]}
+        noButton
+        noCaret
+        title={<i className="fa fa-question-circle-o co-masthead__help-icon" />} />
+      <AboutModal className="co-masthead__modal" logo={okdLogoImg} altLogo="OKD" productTitle="OpenShift" show={this.state.showModal} onHide={this.close}>
+        <h2>About</h2>
+        <p>OpenShift is Red Hat&apos;s container application platform that allows developers to quickly develop, host, and scale applications in a cloud environment.</p>
+        <h2>Version</h2>
+        <SoftwareDetails className="co-masthead__modal--details" kubernetesTitle="Kubernetes Master" noColumns openShiftFirst openShiftTitle="OpenShift Master" titleClassName="co-masthead__modal--subtitle" detailClassName="co-masthead__modal--version" />
+        {username && <React.Fragment>
+          <h2>Account</h2>
+          <p>You are currently logged in under the user account <span className="co-masthead__modal--username">{username}</span>.</p>
+        </React.Fragment>}
+      </AboutModal>
+    </React.Fragment>;
+  }
+}
+
+export class OSHelpMenu extends SafetyFirst<{}, OSHelpMenuState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: undefined
+    };
+  }
+
+  componentDidMount() {
+    super.componentDidMount();
+    this._getUserInfo();
+  }
+
+  _getUserInfo() {
+    coFetchJSON('api/kubernetes/apis/user.openshift.io/v1/users/~')
```

Comment:
It would be better to do this once and keep it in Redux instead of multiple times for different nav components. Then the modal itself can connect to Redux to get the username rather than passing it down.

It might be simpler for now just to not include the username in the modal, however.